### PR TITLE
Update to support MonsterInsights > 6.0 + a few other things

### LINKED ISSUE #14: Google Analytics by Yoast Premium
Would be great if this worked with Google Analytics by Yoast Premium as well. For those of us that have Google Analytics by Yoast Premium installed, this plugin does not seem to work.

### LINKED ISSUE #17: `reset()` is generating a notice
```
PHP Notice:  Only variables should be passed by reference in /var/www/web/app/plugins/google-analytics-top-posts-widget/includes/class-ga-top-content.php on line 467
```

Should be a quick fix, assigning to a variable to pass to `reset()` or just avoiding it, whatever seems better.

### PR DESCRIPTION
**Pull request by chriscct7**
Changes:

- Updated plugin to use the MI 6.0 classes for GA when MI >= 6.0.
- Changed verbiage from login to GA to authenticate with GA to match the MI plugin setting's button label
- Fixes #17 (needed to do a quick test of other changes)
- Fixes #14 for MI >= 6.0
- Added code to ensure correct admin settings page is used when MI >= 6.0
- Changed references from GA by Yoast to GA by MI to reflect the plugin name change

### CODE REVIEW

**Comment by jtsternberg** [comment]
Whoa. You are the man. I'll review this later today. Thanks a ton for being proactive and helpful!

**Comment by chriscct7** [comment]
Not a problem! Sent an email over with the new version + some more detail on things.

**Comment by chriscct7** [comment]
Note this has been minimally tested. More extensive testing is probably not a bad idea.

File: readme.md [hunk 1]
```diff
@@ -14,13 +14,13 @@
...
 
-Widget and shortcode to display top content according to Google Analytics. ("Google Analytics by Yoast" plugin required)
+Widget and shortcode to display top content according to Google Analytics. ("Google Analytics by MonsterInsights" plugin required)
 
 ## Description ##
 
 Adds a widget that allows you to display top pages/posts in your sidebar based on google analytics data.
 
-Requires a Google Analytics account, and the plugin, ["Google Analytics by Yoast"](http://wordpress.org/extend/plugins/google-analytics-dashboard/) (which will be auto-installed by this plugin, thanks to [@jthomasgriffin](http://twitter.com/jthomasgriffin)'s awesome [TGM Plugin Activation Class](http://j.ustin.co/yZPKXw)).
+Requires a Google Analytics account, and the plugin, ["Google Analytics by MonsterInsights"](http://wordpress.org/extend/plugins/google-analytics-dashboard/) (which will be auto-installed by this plugin, thanks to [@jthomasgriffin](http://twitter.com/jthomasgriffin)'s awesome [TGM Plugin Activation Class](http://j.ustin.co/yZPKXw)).
```
Comment (chriscct7) [inline readme.md]: The link here needs to be replaced to the right plugin

**Review by chriscct7 — COMMENTED** [review]
Missed a link to correct

**Comment by chriscct7** [comment]
Noticed I had missed a link correction earlier in the .md and .txt files. This has been corrected.

**Comment by jtsternberg** [comment]
Thanks a ton Chris. This is merged, and i'll be releasing an update to .org soon.

**Pull request merged**